narwhals/_spark_like/dataframe.py

```diff
@@ -112,17 +115,53 @@ def _from_native_frame(self: Self, df: DataFrame) -> Self:
     def columns(self: Self) -> list[str]:
         return self._native_frame.columns  # type: ignore[no-any-return]
 
-    def collect(self: Self) -> PandasLikeDataFrame:
-        import pandas as pd  # ignore-banned-import()
+    def collect(
+        self: Self,
+        backend: ModuleType | Implementation | str | None,
+        **kwargs: Any,
+    ) -> CompliantDataFrame:
+        if backend in (None, "pandas", Implementation.PANDAS, get_pandas()):
+            import pandas as pd  # ignore-banned-import
+
+            from narwhals._pandas_like.dataframe import PandasLikeDataFrame
+
+            return PandasLikeDataFrame(
+                native_dataframe=self._native_frame.toPandas(),
+                implementation=Implementation.PANDAS,
+                backend_version=parse_version(pd.__version__),
+                version=self._version,
+            )
 
-        from narwhals._pandas_like.dataframe import PandasLikeDataFrame
+        elif backend in ("pyarrow", Implementation.PYARROW, get_pyarrow()):
+            import pyarrow as pa  # ignore-banned-import
 
-        return PandasLikeDataFrame(
-            native_dataframe=self._native_frame.toPandas(),
-            implementation=Implementation.PANDAS,
-            backend_version=parse_version(pd.__version__),
-            version=self._version,
-        )
+            from narwhals._arrow.dataframe import ArrowDataFrame
+
+            return ArrowDataFrame(
+                native_dataframe=pa.Table.from_batches(
+                    self._native_frame._collect_as_arrow()
+                ),
```

Comment:
I am defaulting to pandas instead of pyarrow because otherwise some tests will fail. Specifically those that result in an empty dataframe - pyarrow `from_batches` raises that RecordBatches cannot be empty. Maybe once pyspark 4.0 is out this could be better integrated?

Reply:
this would technically be a breaking change - can we catch the exception and return an empty dataframe?

Reply:
Sure I will take a look. Notice that changing to pyarrow now is also a breaking change 🙈

Reply:
oh right we currently collect into pandas? 

🙈

Reply:
we've documented pyspark support as "work-in-progress" so we may not need to feel too guilty about changing this 😄

Reply:
yes correct 🙃